File: 2274-keep-multiplying-found-values-by-two/2274-keep-multiplying-found-values-by-two.cpp
#pragma GCC optimize("O3", "unroll-loops");
static const int __ = []() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    return 0;
}();
class Solution {
   public:
    int findFinalValue(vector<int>& nums, int original) {
        vector<int> freq(2001, 0);
        for (const auto& num : nums) {
            freq[num]++;
        }

        while (freq[original]) {
            original *= 2;
        }

        return original;
    }
};

File: 2274-keep-multiplying-found-values-by-two/2274-keep-multiplying-found-values-by-two.rs
impl Solution {
    pub fn find_final_value(nums: Vec<i32>, original: i32) -> i32 {
        let mut freq = vec![0; 2001];
        let mut ans = original as usize;
        for num in nums {
            freq[num as usize] += 1;
        }

        while freq[ans] > 0 {
            ans *= 2;
        }

        ans as i32
    }
}